### PR TITLE
Add checks for brackets preceded or followed by spaces

### PR DESCRIPTION
**Pull request by fred-atherden**

### CODE REVIEW

File: test/tests/gen/bracket-tests/bracket-test-5/fail.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+Test: report    matches(.,'[\(\[]\p{Zs}+')
+Message:  element contains an opening bracket immediately followed by a space (e.g. '( ' or '[ '). Should the space be removed?  -->
+<root xmlns:ali="http://www.niso.org/schemas/ali/1.0/" xmlns:mml="http://www.w3.org/1998/Math/MathML" xmlns:xlink="http://www.w3.org/1999/xlink">
+<article>
+<p>( <bold>Some</bold> text <italic>in</italic> <xref>parentheses</xref>)</p>
+<td>[          <bold>Some</bold> text <italic>in</italic> <xref>parentheses</xref>]</td>
+<title>(   Dummy</title>
+<title>Text)</title>
+</article>
+</root>
```
Comment (fred-atherden) [inline test/tests/gen/bracket-tests/bracket-test-5/fail.xml:12]: @rkcook312, is this the kind of thing you were talking about earlier (along with [bracket-test-6/fail.xml](https://github.com/elifesciences/eLife-JATS-schematron/pull/663/files#diff-2bd52449ed4b32d0939b67a4b00e38283c5b06e8c746d9138e9621035ffc772c) below)?

Any cases you can think of that are missing?

**Comment by rkcook312** [comment]
Thanks @fred-atherden, both fail tests capture what I was mentioning, all good to me.

**Comment by fred-atherden** [comment]
Great - thanks for taking a look!

**Comment by rkcook312** [comment]
found an example where this has occurred in 96602
Screenshot 2024-07-25 at 11 24 04
Screenshot 2024-07-25 at 11 26 25

**Comment by fred-atherden** [comment]
Nice thanks @rkcook312 - was this (now) flagged via Schematron, or did the check not work for some unknown reason. if the latter, pleasecould you attach the XML so I can look into it further?

**Comment by rkcook312** [comment]
@fred-atherden I've rerun the xml and the test is working now, thanks!

**Comment by fred-atherden** [comment]
Glad to hear it - thanks!

**Pull request merged**